tests/test_2024_04_23.py

```diff
@@ -0,0 +1,132 @@
+
+def _check_no_role(contract: Contract, role: str, holder: str):
+    role_bytes = web3.keccak(text=role).hex()
+    assert not contract.getRoleMember(role_bytes, 0) == holder, f"Role {role} holder on {contract} should be {holder}"
+
+def test_vote(helpers, vote_ids_from_env, bypass_events_decoding, accounts):
+
+    #parameter
+    agent = contracts.agent
+
+    #old GateSeal permissions before
+    _check_role(contracts.withdrawal_queue, "PAUSE_ROLE", old_gate_seal)
+    _check_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", old_gate_seal)
+    #new GateSeal permissions before
+    _check_no_role(contracts.withdrawal_queue, "PAUSE_ROLE", new_gate_seal)
+    _check_no_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", new_gate_seal)
+
+    # START VOTE
+    if len(vote_ids_from_env) > 0:
+        (vote_id,) = vote_ids_from_env
+    else:
+        tx_params = {"from": "0x109b9744397acf987a1abb5bb4eef7362ab9ff66"} #LDO_HOLDER_ADDRESS_FOR_TESTS
+        vote_id, _ = start_vote(tx_params, silent=True)
+
+    vote_tx = helpers.execute_vote(accounts, vote_id, contracts.voting)
+
+    print(f"voteId = {vote_id}, gasUsed = {vote_tx.gas_used}")
+
+    #old GateSeal permissions after
+    _check_no_role(contracts.withdrawal_queue, "PAUSE_ROLE", old_gate_seal)
+    _check_no_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", old_gate_seal)
+    #new GateSeal permissions after
+    _check_role(contracts.withdrawal_queue, "PAUSE_ROLE", new_gate_seal)
+    _check_role(contracts.validators_exit_bus_oracle, "PAUSE_ROLE", new_gate_seal)
+
+    #Асceptance test
+    new_gate_seal_contract = interface.GateSeal(new_gate_seal)
+    assert new_gate_seal_contract.get_sealing_committee() == GATE_SEAL_COMMITTEE
+    sealables = new_gate_seal_contract.get_sealables()
+    assert len(sealables) == 2
+    assert contracts.validators_exit_bus_oracle.address in sealables
+    assert contracts.withdrawal_queue.address in sealables
+
+    assert new_gate_seal_contract.get_seal_duration_seconds() == GATE_SEAL_PAUSE_DURATION
+    assert new_gate_seal_contract.get_expiry_timestamp() == NEW_EXPIRY_TIMESTAMP
+    assert not new_gate_seal_contract.is_expired()
+
+    #Scenario test
+    print(f"Simulating GateSeal flow")
+
+    #Try to use the Old gate seal to pause the contracts
+    print("Try to use the Old gate seal to pause the contracts")
+    print(chain.time())
+    with reverts("10"): # converted into string list of sealed indexes (in sealables) in which the error occurred, in the descending order
+        contracts.gate_seal.seal(sealables, {"from": GATE_SEAL_COMMITTEE})
+
+    print("Seal the contracts with the New gate seal")
+    new_gate_seal_contract.seal(sealables, {"from": GATE_SEAL_COMMITTEE})
+    print("Sealed")
+
+    expiry_timestamp = chain.time()
+    assert new_gate_seal_contract.is_expired()
+    assert expiry_timestamp - new_gate_seal_contract.get_expiry_timestamp() <= 2
```

Comment:
need to change order, to avoid case when new_gate_seal_contract.get_expiry_timestamp() is after expiry_timestamp